### DISCUSSION TITLE
Unmask before submit

### DISCUSSION THREAD

**Discussion by alexandremjacques** [uNmAnNeR/imaskjs#661] [Q&A]
Hi,

I'm using Unpoly framework that allows me to attach a instance of iMask to each element with a selector:

```
up.compiler('.masked', function (elem) {
    IMask(elem, {
        mask: '000.000.000-00'
    })
})
```

Before submitting the form (Unpoly hooks an event handler that allows me doing something before submit), I need to unmask all masked inputs. Is there a way to traverse the DOM or the form elements to find  masked inputs and set the .unmaskedValue on them?

**Comment by gormonn** [comment]
@alexandremjacques A similar issue is found when using iMask with react-hook-form.
It seems you can use https://imask.js.org/guide.html#pipe for this
